src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/Microservice.kt

```diff
@@ -256,9 +255,9 @@ class Microservice(
 
         val rawMessage = message.rawMessage
 
-        rawMessage.eventId?.run {
-            if (!bookNames.contains(bookName)) {
-                onHandleError("Unexpected book name: $bookName (expected one of: ${bookNames})", message)
+        rawMessage.eventId?.also { eventId ->
+            if (!bookNames.contains(eventId.bookName)) {
+                onHandleError("Unexpected book name: ${eventId.bookName} (expected one of: ${bookNames})", message)
```

Comment:
question: What was wrong with old version? `bookName` in both cases should be resolved to EventID bookName property

Reply:
Here nothing, but transport group handling was broken. I rewrote this part to clarify the code.